SignalServiceKit/src/Util/AppReadiness.m

```diff
@@ -94,10 +94,12 @@ - (void)runAppReadyBlocks
     OWSAssertIsOnMainThread();
     OWSAssert(self.isAppReady);
 
-    for (AppReadyBlock block in self.appReadyBlocks) {
+    // Make a local copy, then clear this state.
+    NSArray<AppReadyBlock> *appReadyBlocks = self.appReadyBlocks;
+    self.appReadyBlocks = nil;
```

Comment:
A third fix.  Again, this does not address a known issue, but it's safer.

This makes us robust to enqueuing a new "app ready block" while executing the existing blocks.